src/test/java/org/springframework/data/hazelcast/repository/query/QueryIT.java

```diff
@@ -831,4 +823,56 @@ public void findByLastnameListenableFuture()
         assertThat(result.get(), containsInAnyOrder(person1, person2));
     }
     
+    @Test
+    public void countDistinctByFirstname() {
+    	//adding a duplicate Person with different id
+    	Person p = new Person();
+    	p.setId("1970");
+    	p.setFirstname("George");
+    	p.setLastname("Scott");
+    	this.personMap.put("1001", p);
+    	
+        final Long distinctPersons = this.personRepository.countDistinctByFirstname("George");
+        assertThat( distinctPersons, equalTo(2L));
+		this.personMap.remove("1001");
+    }
```

Comment:
Ideally, these tests should create two totally new entries, and then perform logic on it.

Now, these tests are stateful and it would be great if we could not rely on the previous state.